Portchain/stream-sea-client-node
# Reconnect after dropped connections

With this PR, stream-sea-client reconnects after a connection is dropped.

This PR refactors the WSClient class into four classes, each of which extends EventEmitter:
StreamSeaClient
StreamSeaConnection
StreamSeaSocket
StreamSeaSubscription

It also adds unit tests for StreamSeaClient and StreamSeaConnection.

### Manual testing

- Link stream-sea-cli against stream-sea-client-node with this PR.
- Subscribe to a stream S in one CLI session C1.
- Publish a message M to S in another CLI session C2.
- Verify M is printed to the console in session C1.
- Disable internet access
- Verify that the client in C1 eventually starts reconnection attempts.
- Enable internet access
- Verify that the client in C1 successfully reconnects.
- Publish a message M to S in another CLI session C2.
- Verify M is printed to the console in session C1.
